lang/rust/avro/src/types.rs

```diff
@@ -333,8 +333,15 @@ impl Value {
     /// See the [Avro specification](https://avro.apache.org/docs/current/spec.html)
     /// for the full set of rules of schema validation.
     pub fn validate(&self, schema: &Schema) -> bool {
+        let rs = ResolvedSchema::try_from(schema).expect("Schema didn't successfully parse");
+        self.validate_internal(schema, rs.get_names())
+    }
+
+    fn validate_internal(&self, schema: &Schema, names: &NamesRef) -> bool {
```

Comment:
I believe we will need the enclosing namespace passed through here as well to help with any Name Refs that don't include full namespaces. Will check.

Reply:
I think the SchemaRef's `Name` already brings the full information!